opm/simulators/timestepping/AdaptiveTimeStepping.hpp

```diff
@@ -426,24 +426,37 @@ void registerAdaptiveParameters();
                             // Found no wells to close, chop the timestep as above.
                             chopTimestep();
                         } else {
-                            // Close all consistently failing wells.
+                            // Close all consistently failing wells that are not under group control
                             int num_shut_wells = 0;
+                            std::set<std::string> shut_wells;
```

Comment:
since you only need `insert` and  loop over it, `std::vector` will be a better choice.